src/main/java/graphql/annotations/GraphQLInputObjectRetriever.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2016 Yurii Rashkovskii
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ */
+package graphql.annotations;
+
+
+import graphql.schema.*;
+import graphql.schema.GraphQLNonNull;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class GraphQLInputObjectRetriever {
+
+    public GraphQLInputType getInputObject(graphql.schema.GraphQLType graphQLType, String newNamePrefix, Map<String, graphql.schema.GraphQLType> typeRegistry) {
+        if (graphQLType instanceof GraphQLObjectType) {
+            GraphQLObjectType object = (GraphQLObjectType) graphQLType;
+            if (typeRegistry.containsKey(newNamePrefix + object.getName()) && typeRegistry.get(newNamePrefix + object.getName()) instanceof GraphQLInputType) {
```

Comment:
newNamePrefix + object.getName() -> to a variable

Reply:
also you can extract the typeRegistry.get to a variable and check if not null and use only this var (what you prefer)